include/aikido/robot/Hand.hpp

```diff
@@ -0,0 +1,109 @@
+#ifndef AIKIDO_ROBOT_HAND_HPP_
+#define AIKIDO_ROBOT_HAND_HPP_
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <dart/dart.hpp>
+#include "aikido/robot/GrabInfo.hpp"
+#include "aikido/io/yaml.hpp"
+#include "aikido/control/PositionCommandExecutor.hpp"
+
+namespace aikido {
+namespace robot {
+
+/// A base class for a hand used by manipulators
+class Hand
+{
+public:
+
+  /// \param newName New name for this robot
+  virtual std::unique_ptr<Hand> clone(const std::string& newName) override;
+
+  /// Get the end-effector body node.
+  /// \return DART body node of end-effector
+  dart::dynamics::BodyNode* getBodyNode() const;
```

Comment:
This is not const-correct. `dart::dynamics::BodyNode* getBodyNode() const` is a const function so the caller would assume that calling this function doesn't change this `Hand` instance. However, since this function returns non-const `BodyNode` pointer, this function can be changed because the returned `BodyNode` can be changed.

So, to solve this issue, we might want to have two functions like
```cpp
dart::dynamics::BodyNode* getBodyNode();
const dart::dynamics::BodyNode* getBodyNode() const;
```

If this function returned a `shared_ptr`, then it should be like:
```cpp
std::shared_ptr<dart::dynamics::BodyNode> getBodyNode();
std::shared_ptr<const dart::dynamics::BodyNode> getBodyNode() const;
```
Note that the last line shouldn't be:
```cpp
const std::shared_ptr<dart::dynamics::BodyNode>& getBodyNode() const;
```
because it still returns non const pointer (but just const `shared_ptr` object).